dashboard/test/lib/policies/child_account_test.rb

```diff
@@ -496,73 +497,37 @@ class ComplianceStateTest < ActiveSupport::TestCase
     end
   end
 
-  describe '.personal_account?' do
-    let(:personal_account?) {Policies::ChildAccount.personal_account?(user)}
-
-    let(:user_sponsored?) {false}
-    let(:user_migrated?) {false}
-    let(:user_provider) {'email'}
-    let(:user_auth_option_credential_type) {'email'}
-    let(:user_auth_option) {build_stubbed(:authentication_option, credential_type: user_auth_option_credential_type)}
-    let(:user) {build_stubbed(:user, provider: user_provider, authentication_options: [user_auth_option])}
-
-    before do
-      user.stubs(:sponsored?).returns(user_sponsored?)
-      user.stubs(:migrated?).returns(user_migrated?)
-    end
-
-    it 'returns true' do
-      _(personal_account?).must_equal true
-    end
-
-    context 'when user provider is Clever' do
-      let(:user_provider) {'clever'}
-
-      it 'returns false' do
-        _(personal_account?).must_equal false
-      end
-    end
-
-    context 'when user provider is LTI v1' do
-      let(:user_provider) {'lti_v1'}
-
-      it 'returns false' do
-        _(personal_account?).must_equal false
-      end
-    end
-
-    context 'when user is migrated' do
-      let(:user_migrated?) {true}
-      let(:user_provider) {'clever'}
-
-      it 'returns true' do
-        _(personal_account?).must_equal true
-      end
-
-      context 'when credential type of user auth option is Clever' do
-        let(:user_auth_option_credential_type) {'clever'}
-
-        it 'returns false' do
-          _(personal_account?).must_equal false
-        end
-      end
-
-      context 'when credential type of user auth option is LTI v1' do
-        let(:user_auth_option_credential_type) {'lti_v1'}
-
-        it 'returns false' do
-          _(personal_account?).must_equal false
-        end
-      end
-    end
-
-    context 'when user is sponsored' do
-      let(:user_sponsored?) {true}
-
-      it 'returns false' do
-        _(personal_account?).must_equal false
-      end
+  test '.personal_account?' do
+    # [User traits, Expected result from personal_account?]
+    test_matrix = [
```

Comment:
Dang this test matrix is getting big. I'm wondering if there is an easier way to organize it so it's harder to miss test cases. 

I think this matrix is missing tests for when a user has two auth options, like Email + Clever, Email + Google + Section. Basically the test that Email always returns `true` despite any other settings.

Reply:
Good point re: `:without_email_auth_option`. I'll add it into the matrix. The way `user_predates_policy?` is currently implemented, it doesn't matter because the function short-circuits before it checks the auth options if the created_at date is prior to the policy coming into effect. But it's still good to include all the correct traits to ensure these tests continue to do what we want them to.